### PR TITLE
[fix] Convert attached_policies_arns into a map within the aws_iam_role module

### PR DESCRIPTION
**Pull request by michaelmcchen**
### Summary
In situations where you were appending new resources to `attached_policies_arns` that didn't exist yet, Terraform would throw the following error:
```
Error: Invalid for_each argument

  on .terraform/modules/aws-role-devserver-restart/aws-iam-role/main.tf line 41, in resource "aws_iam_role_policy_attachment" "policy_attachment":
  41:   for_each   = toset(var.attached_policies_arns)

The "for_each" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the for_each depends on.
```
This PR replaces the list with a map so that Terraform can get the size statically during the plan. 

### Test Plan
Updating the consuming module in https://github.com/FB-PLP/terraform-infra-management/pull/900 and TF plan ([here](https://si.prod.tfe.czi.technology/app/lp-infra/accounts-summitps/runs/run-6pFATci8iJu1V6fh)) seems as expected 

### References
(Optional) Additional links to provide more context.

### CODE REVIEW

File: aws-iam-role/variables.tf [hunk 1]
```diff
@@ -52,8 +52,8 @@ variable inline_policies {
   default     = []
 }
 
-variable attached_policies_arns {
-  type        = list(string)
-  description = "List of policies ARNs to be attached to the IAM role."
-  default     = []
+variable attached_policies_names_arns {
+  type        = map
```
Comment (edulop91) [inline aws-iam-role/variables.tf:56]:
```suggestion
  type        = map(string)
```

File: aws-iam-role/variables.tf [hunk 2]
```diff
@@ -52,8 +52,8 @@ variable inline_policies {
...
+  description = "Map of policy names to the respective ARNs to be attached to the IAM role. "
```
Comment (edulop91) [inline aws-iam-role/variables.tf:57]:
```suggestion
  description = "Map of policy names to the respective ARNs to be attached to the IAM role."
```

File: aws-iam-role/README.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ module iam-role {
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
-| attached\_policies\_arns | List of policies ARNs to be attached to the IAM role. | `list(string)` | `[]` | no |
+| attached\_policies\_names\_arns | Map of policy names to the respective ARNs to be attached to the IAM role. | `map` | `{}` | no |
```
Comment (Dingying0410) [inline aws-iam-role/README.md:45]: overall lgtm, (non-blocking) this requires ppl to input the unnecessary names though, I wonder if we can make this more straightforward by filling out the map itself in the module, so ppl still provides the list in the module and the module converts the list into a map. but overall, current version lgtm!

Comment (Dingying0410) [inline aws-iam-role/README.md:45]:
```suggestion
| attached\_policies\_names\_arns | Map of policy names to the respective ARNs to be attached to the IAM role. | `map(string)` | `{}` | no |
```

**Review by Dingying0410 — APPROVED** [review]

**Review by ryanking — APPROVED** [review]

File: aws-iam-role/README.md (hunk 3 above)
Comment (michaelmcchen) [reply aws-iam-role/README.md:45]: Yeah that's a good idea, thanks! I'll do that

File: aws-iam-role/main.tf [hunk 4]
```diff
@@ -1,4 +1,8 @@
 locals {
+  // Slight hack so Terraform can get the size statically during the plan.
```
Comment (cziie) [inline aws-iam-role/main.tf:2]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Single line comments should begin with #

File: aws-iam-role/main.tf [hunk 5]
```diff
@@ -1,4 +1,8 @@
 locals {
+  // Slight hack so Terraform can get the size statically during the plan. 
+  // Simply passing the list to `for_each` throws an Invalid for_each argument
```
Comment (cziie) [inline aws-iam-role/main.tf:3]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Single line comments should begin with #

File: aws-iam-role/main.tf (hunk 4 above)
Comment (michaelmcchen) [reply aws-iam-role/main.tf:2]: :'(

**Pull request merged**